src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -80,13 +83,77 @@ def _print_extensions_msg(help_file):
             if help_file.command_source.preview:
                 logger.warning(help_file.command_source.get_preview_warn_msg())
 
-    def print_detailed_help(self, cli_name, help_file):
+    def _print_detailed_help(self, cli_name, help_file):
         AzCliHelp._print_extensions_msg(help_file)
-        self._print_detailed_help(cli_name, help_file)
+        super(AzCliHelp, self)._print_detailed_help(cli_name, help_file)
+
+    @staticmethod
+    def update_parser_with_help_file(nouns, cmd_loader_map, parser, is_group):
+        import inspect
+        import os
+
+        def _parse_yaml_from_string(text, help_file_path):
+            import yaml
+
+            dir_name = os.path.dirname(help_file_path)
+            base_name = os.path.basename(help_file_path)
+
+            pretty_file_path = os.path.join(os.path.basename(dir_name), base_name)
+
+            try:
+                data = yaml.load(text)
+                if not data:
+                    raise CLIError("Error: Help file {} is empty".format(pretty_file_path))
+                return data
+            except yaml.YAMLError as e:
+                raise CLIError("Error parsing {}:\n\n{}".format(pretty_file_path, e))
+
+        command_nouns = " ".join(nouns)
+        loader = None
+        if is_group:
+            for k, v in cmd_loader_map.items():
+                if k.startswith(command_nouns):
+                    loader = v[0]
+                    break
+        else:
+            loader = cmd_loader_map.get(command_nouns, [None])[0]
+
+        if loader:
+            loader_file_path = inspect.getfile(loader.__class__)
+            dir_name = os.path.dirname(loader_file_path)
+            files = os.listdir(dir_name)
+            for file in files:
+                if file.endswith(".yaml") or file.endswith(".yml"):
+                    help_file_path = os.path.join(dir_name, file)
+                    with open(help_file_path, "r") as f:
+                        text = f.read()
+                        data = _parse_yaml_from_string(text, help_file_path)
+                        parser.help_file_data = data
+                        return
+
+    def show_help(self, cli_name, nouns, parser, is_group):
+        cmd_loader_map_ref = self.cli_ctx.invocation.commands_loader.cmd_to_loader_map
+        self.update_parser_with_help_file(nouns, cmd_loader_map_ref, parser, is_group)
+        super(AzCliHelp, self).show_help(cli_name, nouns, parser, is_group)
 
 
 class CliHelpFile(KnackHelpFile):
 
+    GROUP_TYPE = "group"
+    COMMAND_TYPE = "command"
+    CONTENT_TYPES = [COMMAND_TYPE, GROUP_TYPE]
+
+    def load(self, options):
+
+        if hasattr(options, "help_file_data"):
+            data = self._load_from_parsed_yaml(options.help_file_data)
+            if "content" not in data:
+                self._load_from_data(data)
```

Comment:
Is this the "old style" loading?  If so, probably add a comment.

Reply:
This is actually the new style of loading. But I added a comment to explain. The idea is that if we are able to properly transform the help.yaml schema to the expected schema, then we can actually update the help file object. Which is why we call `self._load_from_data()` and not `super().load()`